ams/CleanupCommand.cs

```diff
@@ -0,0 +1,226 @@
+            }
+            else
+            {
+                table.AddColumn("Asset");
+            }
+            table.AddColumn("IsDeleted");
+            foreach (var (key, value) in stats)
+            {
+                var status = value ? $"[green]{value}[/]" : $"[red]{value}[/]";
+                table.AddRow($"[green]{key}[/]", status);
+            }
+
+            _console.Write(table);
+        }
+        private async Task<bool> CleanUpAccountAsync(MediaServicesAccountResource account, CancellationToken cancellationToken)
+        {
+            try
+            {
+
+                var deleteOperation = await account.DeleteAsync(WaitUntil.Completed);
+
+                if (deleteOperation.HasCompleted && deleteOperation.GetRawResponse().Status == 200)
+                {
+                    _logger.LogInformation("The media account {account} has been deleted.", account.Data.Name);
+                    return true;
+                }
+                else
+                {
+                    _logger.LogInformation("The media account {account} deletion failed.", account.Data.Name);
+                    return false;
+                }
+
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to delete account {name}", account.Data.Name);
+                return false;
+            }
+        }
+        private async Task<bool> CleanUpAssetAsync(MediaServicesAccountResource account, MediaAssetResource asset, CancellationToken cancellationToken)
+        {
+            try
+            {
+
+                var storage = await _resourceProvider.GetStorageAccountAsync(account, cancellationToken);
+                var container = storage.GetContainer(asset);
+                if (!await container.ExistsAsync(cancellationToken))
+                {
+                    _logger.LogWarning("Container {name} missing for asset {asset}", container.Name, asset.Data.Name);
+
+                    return false;
+                }
+
+                // The asset container exists, try to check the metadata list first.
+                var migrateResult = await _tracker.GetMigrationStatusAsync(container, cancellationToken);
+
+                if (_options.IsCleanUpAccount || _options.IsForceCleanUpAsset || migrateResult.Status == MigrationStatus.Completed)
+                {
+                    var policies = account.GetContentKeyPolicies();
+                    var locator = await account.GetStreamingLocatorAsync(asset, cancellationToken);
+                    if (_options.IsCleanUpAccount)
+                    {
+                        var endpoints = account.GetStreamingEndpoints();
+                        var liveevents = account.GetMediaLiveEvents();
+                        await CleanUpContentAsync(container, asset, locator, endpoints, policies, liveevents);
+                        _logger.LogDebug("account {account} is deleted.", account.Data.Name);
```

Comment:
When cleaning up a specific Asset, there is no need to touch StreamingEndpoint and LiveEvents.  the clean up of StreamingEndpoint and LiveEvents should be moved to the CleanupAccount method.